File: c/assignment7.c
// C program to reverse a string using two pointers

#include <stdio.h>
#include <string.h>

char* reverseString(char *s) {
    int left = 0, right = strlen(s) - 1;

    // Swap characters from both ends till we reach
    // the middle of the string
    while (left < right) {
        char temp = s[left];
        s[left] = s[right];
        s[right] = temp;
        left++;
        right--;
    }
  
    return s;
}

int main() {
    char s[] = "abdcfe"; 
    printf("%s", reverseString(s)); 
    return 0;
}

File: python/assignment7.py
# Python program to reverse a string using two pointers

# Function to reverse a string using two pointers

def reverseString(s):
    left = 0
    right = len(s) - 1
    
    # Convert string to a list for mutability
    s = list(s)  
    
    # Swap characters from both ends till we reach
    # the middle of the string
    while left < right:
        s[left], s[right] = s[right], s[left]
        left += 1
        right -= 1
    
    # Convert list back to string
    return ''.join(s)  

if __name__ == "__main__":
    s = "abdcfe"
    print(reverseString(s))